docs/en/operations/configuration-files.md

```diff
@@ -54,6 +54,42 @@ XML substitution example:
 
 Substitutions can also be performed from ZooKeeper. To do this, specify the attribute `from_zk = "/path/to/node"`. The element value is replaced with the contents of the node at `/path/to/node` in ZooKeeper. You can also put an entire XML subtree on the ZooKeeper node and it will be fully inserted into the source element.
 
+## Decryption {#decryption}
+
+Elements with text nodes may be encrypted with [encryption codecs](../sql-reference/statements/create/table.md#encryption-codecs). In this case `<encryption_codecs>` section should be included in configuration file and each element node with encrypted text should have `encryption_codec` attribute with name of codec.
+
+Example:
+
+```xml
+<clickhouse>
+    <encryption_codecs>
+        <aes_128_gcm_siv>
+            <key_hex>00112233445566778899aabbccddeeff</key_hex>
+        </aes_128_gcm_siv>
+    </encryption_codecs>
+    <interserver_http_credentials>
+        <user>admin</user>
+        <password  encryption_codec="AES_128_GCM_SIV">961F000000040000000000EEDDEF4F453CFE6457C4234BD7C09258BD651D85</password>
+    </interserver_http_credentials>
+</clickhouse>
+```
+
+To get the encrypted value `encrypt_decrypt` example application may be used.
```

Comment:
Related to that, it would be nice to include the steps in the documentation how the encrypted password (`961F000000040000...`) is obtained.

Reply:
@rschu1ze I have spent few hours in trying to encrypt/decrypt using Python. 
It's not just pseudo code:
```
encrypted_text=encrypt(plain_text)
plain_text=decrypt(encrypted_text)
```
Because there is a special header created in ClickHouse code. 
It's like:
```
encrypted_text=encrypt(header + plain_text)
```
and `encrypted_text` also has own header.

Read more about the header:
https://github.com/ClickHouse/ClickHouse/blob/10b6fc9237327e867a1ef23f54bfcfc9318c4db8/src/Compression/CompressionInfo.h#L9

So it's not a trivial task for Python programmer to implement encrypt/decrypt.

Furthermore:
I searched for AES-GCM-SIV implementation in Python. I found the [one](https://github.com/Shay-Gueron/AES-GCM-SIV) which is not in production level and it's not quite popular.

The popular modules in Python's worlds are:
1. [pycryptodome](https://pycryptodome.readthedocs.io/en/latest/src/cipher/modern.html#gcm-mode)
2. [cryptography](https://cryptography.io/en/latest/hazmat/primitives/aead/#cryptography.hazmat.primitives.ciphers.aead.AESGCM)
But they both support only AES-GCM. And AES-GCM is different from AES-GCM-SIV  because it uses another hashing function. 

So I propose to keep `encrypt_decrypt` at least in examples. What do you think?

Reply:
OK. We will leave it in examples in current PR. In future it could be moved to utils like `config-processor `or other place.